transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpSubjectSecurityFilter.java

```diff
@@ -112,6 +119,39 @@ public void doMessageReceived(NextFilter nextFilter, IoSession session, Object m
         if (! httpRequestMessageReceived(nextFilter, session, message)) return;
 
         HttpRequestMessage httpRequest = (HttpRequestMessage) message;
+        final boolean loggerIsEnabled = logger != null && logger.isTraceEnabled();
+
+        if (httpRequest.getHeader(HEADER_X_FORWARDED_FOR) == null) {
+            String remoteIpAddress = null;
+            SocketAddress socketAddress = session.getRemoteAddress();
+
+            if (socketAddress instanceof InetSocketAddress) {
+                InetAddress addr = ((InetSocketAddress) socketAddress).getAddress();
+                remoteIpAddress = addr.getHostAddress();
+
+                if (loggerIsEnabled) {
+                    logger.trace(format("HttpSubjectSecurityFilter: Remote IP Address: '%s'", remoteIpAddress));
+                }
+            }
+            else if (socketAddress instanceof ResourceAddress) {
+                ResourceAddress resourceAddress = (ResourceAddress) socketAddress;
```

Comment:
Recommend removing the `if` block above, just keep this `else` block and use `REMOTE_ADDRESS.get(session)` instead.